test/Extensions.Mcp.EndToEndTests/Fixtures/McpEndToEndFixtureBase.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Extensions.Mcp.EndToEndTests.Abstractions;
+using Extensions.Mcp.EndToEndTests.AbstractionOverCoreTools;
+using Microsoft.Extensions.Logging;
+using ModelContextProtocol.Client;
+using ModelContextProtocol.Protocol;
+
+// Prevent multiple instances of Core Tools from interfering with one another.
+// If any new collection definitions are added, they should have parallelization disabled.
+[assembly: CollectionBehavior(CollectionBehavior.CollectionPerAssembly)]
+
+namespace Extensions.Mcp.EndToEndTests.Fixtures
+{
+    public abstract class McpEndToEndFixtureBase : CoreToolsProjectBase
+    {
+        private List<IMcpClient> _clients = [];
+
+        protected McpEndToEndFixtureBase(EndToEndTestProject project) : base(project) { }
+
+        public async Task<IMcpClient> CreateClientAsync(HttpTransportMode transportMode = HttpTransportMode.AutoDetect,
+            McpClientOptions? clientOptions = null,
+            ILoggerFactory? loggerFactory = null,
+            DelegatingHandler? delegatingHandler = null,
+            Func<JsonRpcNotification, CancellationToken, ValueTask>? notificationHandler = null)
+        {
+            var transportOptions = new SseClientTransportOptions()
+            {
+                Endpoint = GetEndpointForTransport(transportMode),
+                TransportMode = transportMode,
+                Name = $"TestClient-{transportMode}"
+            };
+
+            SseClientTransport transport;
+
+            if (delegatingHandler is not null)
+            {
+                HttpClient httpClient = new HttpClient(delegatingHandler);
+                transport = new SseClientTransport(transportOptions, httpClient, loggerFactory, true); // letting disposal be handled by the IMcpClient
+            }
+            else
+            {
+                transport = new SseClientTransport(transportOptions, loggerFactory);
+            }
+
+            var client = await McpClientFactory.CreateAsync(transport, clientOptions);
+
+            _clients.Add(client);
+
+            return client;
+        }
+
+        private const string _mcpEndpointRelativePath = "/runtime/webhooks/mcp";
+        private const string _sseRelativePath = $"{_mcpEndpointRelativePath}/sse";
```

Comment:
Do we have a wait to test other transports like the legacy one? (and potential new ones?)